### DISCUSSION TITLE
Pressure Control

### DISCUSSION THREAD

**Discussion by PleaseHelpMeThanks** [e2nIEE/pandapipes#590] [Q&A] [answered]
I'm trying to implement a throttle that reduces the pressure. For that i wanted to use the pressure_control component. However this causes a convergent error. To study the component i created a really simple network. I found out it only allows me to induce a small pressure drop but even with that the massflow changes drastically and doesn't stay constant. Also I've tried different solvers.

Does anybody know why that happens? Or do I have a fundamental wrong understanding about how the pressure_control is meant to be used. I'd be happy for any help or ideas. This is the code in question:

import pandapipes as pp
#create empty net
net = pp.create_empty_network(fluid="hydrogen")

#create junctions
j1 = pp.create_junction(net, pn_bar=2, tfluid_k=293.15, name="Junction 1")
j2 = pp.create_junction(net, pn_bar=2, tfluid_k=293.15, name="Junction 2")
j3 = pp.create_junction(net, pn_bar=1.9, tfluid_k=293.15, name="Junction 3")
j4 = pp.create_junction(net, pn_bar=1.9, tfluid_k=293.15, name="Junction 4")

#create junction elements
ext_grid = pp.create_ext_grid(net, junction=j1, p_bar=2.0, t_k=293.15, name="Grid Connection")
sink = pp.create_sink(net, junction=j4, mdot_kg_per_s=0.045, name="Sink")
press_control = pp.create_pressure_control(net, from_junction=j2, to_junction=j3,
                                                    controlled_junction=j3, controlled_p_bar=1.98,
                                                    name="Pressure Control")

#create branch elements
p1 = pp.create_pipe_from_parameters(net, from_junction=j1, to_junction=j2, length_km=0.1, diameter_m=0.1, name="Pipe")
p2 = pp.create_pipe_from_parameters(net, from_junction=j2, to_junction=j3, length_km=0.1, diameter_m=0.1, name="Pipe")
p3 = pp.create_pipe_from_parameters(net, from_junction=j3, to_junction=j4, length_km=0.1, diameter_m=0.1, name="Pipe")

pp.pipeflow(net, mode="all", solver_method="nr")
print(net.res_junction)
print(net.res_pipe)

**Comment by dlohmeier** [comment, marked as answer]
Dear @PleaseHelpMeThanks ,
thanks for reaching out on this problem. I am not exactly sure what kind of problem you are trying to address, but an important aspect of your setup is that you create a throttle in parallel to a pipe (from junction 1 to 2). This leads to the effect that the pressure at the outlet node of the throttle and the pipe (junction 2) must equal, i.e. the mass flow through the throttle is regulated such that the pressure drop inside the throttle is exactly the same as the pressure drop induced by friction from the mass flow through the parallel pipe. This system has a strong inter-dependency. 
As I said, I am not exactly sure what type of problem you are trying to solve, maybe you should give a short description for us to be able to help you find the correct setup. Also, please note that the pipeflow mode "all" is not implemented correctly for gas networks (the temperature influence on density is too high to create a correct model).
Kind regards!

**Reply by PleaseHelpMeThanks** [reply]
Dear @dlohmeier,

thanks a lot for your quick response. I wasn't aware that the pressure control creates a new pipe. I thought it was controlling the existing pipe. Knowing that the issue is fixed. 

Thanks again for your help and kind regards!
